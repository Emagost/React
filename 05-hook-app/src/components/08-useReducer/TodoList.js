import React from 'react'
import TodoListItems from './TodoListItems';

const TodoList = ({todos,handleDelete, handleToggle}) => {
    return (
                    <ul className="list-group list-group-flush">
                        {
                            todos.map( (todo, i) => (
                                <TodoListItems
                                    key={todo.id}
                                    todo ={ todo }
                                    i = { i }
                                    handleToggle={ handleToggle }
                                    handleDelete={ handleDelete }
                                />
                            ))
                        }
                    </ul>
    )
}

export default TodoList;
