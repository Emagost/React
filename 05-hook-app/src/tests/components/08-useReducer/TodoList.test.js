import TodoList from "../../../components/08-useReducer/TodoList";
import {shallow} from "enzyme";
import { demoTodos } from "../fixtures/demoTodos";

describe('test <TodoList/>', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(
    <TodoList
        todos = {demoTodos}
        handleDelete = {handleDelete}
        handleToggle = {handleToggle}
    />)

    test('snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('should have two TODOitem', () => {
        
        expect(wrapper.find('TodoListItems').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItems').at(0).prop('handleDelete')).toEqual(expect.any(Function));
        expect(wrapper.find('TodoListItems').at(0).prop('handleToggle')).toEqual(expect.any(Function));

    });
    

});
