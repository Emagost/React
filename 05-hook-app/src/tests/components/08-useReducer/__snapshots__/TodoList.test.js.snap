// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <TodoList/> snapshot 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItems
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "Learn React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItems
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "Learn Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
